app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/main/MainActivity.kt

```diff
@@ -0,0 +1,50 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.main
+
+import android.os.Bundle
+import android.support.v7.widget.GridLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.Window
+import com.naxtlevelofandroiddevelopment.marvelgallery.BuildConfig
+import com.naxtlevelofandroiddevelopment.marvelgallery.R
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.character.CharacterProfileActivity
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common.PresenterBaseActivity
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common.addOnTextChangedListener
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common.bindToSwipeRefresh
+import com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common.toast
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : PresenterBaseActivity(), MainView {
+
+    override var refresh by bindToSwipeRefresh(R.id.swipeRefreshView)
+    override val presenter: MainPresenter by lazy { MainPresenter(this) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        setContentView(R.layout.activity_main)
+        recyclerView.layoutManager = GridLayoutManager(this, 2) as RecyclerView.LayoutManager
+        swipeRefreshView.setOnRefreshListener { presenter.onSearchChanged(searchView.text.toString()) }
+        searchView.addOnTextChangedListener { newText -> presenter.onSearchChanged(newText) }
+    }
+
+    override fun show(items: List<MarvelCharacter>) {
+        val categoryItemAdapters = items.map(this::createCategoryItemAdapter)
+        recyclerView.adapter = MainListAdapter(categoryItemAdapters)
+    }
+
+    override fun showError(error: Throwable) {
+        toast("Error: ${error.message}")
+        if (BuildConfig.DEBUG) {
```

Comment:
It is differently overkill to set up Timber for this purpose. "[Application] should not be complex and it should use only simple and standard dependencies.". But I can delete it, because it is actually open-source project and it is not a problem that logs are printed to user console.